edgelet/provisioning/src/provisioning.rs

```diff
@@ -265,18 +265,24 @@ where
                                     || {
                                         info!(
                                             "A key is expected in the response with the 'symmetric-key' authentication type and the 'source' set to 'payload'.");
-                                        Err(Error::from(ErrorKind::Provision))
+                                        Err(Error::from(ErrorKind::ExternalProvisioning(ExternalProvisioningErrorReason::SymmetricKeyNotSpecified)))
                                     },
                                     |key| {
+                                        let decoded_key = base64::decode(&key).map_err(|_| {
+                                            Error::from(ErrorKind::ExternalProvisioning(ExternalProvisioningErrorReason::InvalidSymmetricKey))
+                                        })?;
+
+                                        key_activator
+                                            .activate_identity_key(KeyIdentity::Device, "primary".to_string(), &decoded_key)
```

Comment:
So, is activation of the key something that was missed in the implementation before?

Reply:
That's correct. Initially, the way we had spec'd the external provisioning mode, it was being assumed that the HSM will have the primary key and the 'activation' of the primary key won't be required as the customer would be creating their own custom HSM library wherein the key would be pre-activated.

However, later we added support for specifying the primary key in the payload itself and didn't think about the 'activation' scenario in that case .

Reply:
I see. Got it.